src/bentoml/_internal/marshal/dispatcher.py

```diff
@@ -351,10 +351,13 @@ async def controller(self):
                 now = time.time()
                 w0 = now - self._queue[0][0]
                 wn = now - self._queue[-1][0]
-                a = self.optimizer.o_a
-                b = self.optimizer.o_b
+                # bound optimizer values at 0; negative values may cause optimizer weirdness
+                a = max(self.optimizer.o_a, 0)
+                b = max(self.optimizer.o_b, 0)
```

Comment:
I think `prevent possible dispatcher bad behavior` mainly means these 2 lines right? The added while loop below is mainly for better code flow?

Reply:
No, the loop below also changes the logic very slightly; it prevents any requests from being canceled once we get into the waiting phase.

Reply:
Removed these, they were being done by the optimizer anyway.